### ISSUE TITLE
auto-format formatting code after comments

### ISSUE THREAD

**Issue by twistedkento** [vim/vim#13374]
### Steps to reproduce

1. I've created a file with the content
```golang
package stupidname

// this is a really long text! this is a really long text! this is a really long text! this is a really long text! 
var exampleText = "This is an example text"
```
2. I open the file with `vim --clean <filename>` and set `fo=acq` and `tw=80`
3. Edit the comment by adding the character `l` to the end
4. The formatter thinks that the variable is a comment paragraph and appends the variable to the end of the comment like this.
```golang
package stupidname

// this is a really long text! this is a really long text! this is a really long
// text! this is a really long text! l var exampleText = "This is an example
// text"
```
5. When I fix the text, so that the `var exampleText = "This is an example text"`is on its own line again, and try to edit the end of the last comment line the variable code again gets appended to the end of the comment paragraph. (sometimes it's enough to just to go up from the code line with the insert cursor to trigger this behavior) Undoing at this stage usually (always in the instance where just moving the insert cursor get auto-format to trigger) results in an version of the text that has the code on one line and still have it at the end of the comment paragraph like this:
```golang
package stupidname

// this is a really long text! this is a really long text! this is a really long
// text! this is a really long text! l var exampleText = "This is an example
var exampleText = "This is an example
// text"
```
6. Bonus info: If I have addons enabled vim is generating multiple error messages (don't know if this is a side effect of this behavior/bug or if YouCompleteMe is just incompatible with auto-format) see the error messages in the log section.

### Expected behaviour

My expectation is for vim to not auto-format the code into a comment section when I have both `a`and `c`flag in the formatoptions settings.
Though I fully understand that maybe the auto-format is not that advanced in understanding go code but I expected it to follow the default `comments` settings.
And the part where it "steals" the code line below when appending text (or even move the insert cursor sometimes) to the comment and the undo artifacts is something that I don't expect the editor to do.

### Version of Vim

9.0.2010 (package gvim from pacman)

### Environment

Operating system: Arch Linux
Terminal: st
Shell: zsh

### Logs and stack traces

```shell
E340: Internal error; if you can reproduce please report a bug
E340: Internal error; if you can reproduce please report a bug
E685: Internal error: text property above joined line not found
E340: Internal error; if you can reproduce please report a bug
E340: Internal error; if you can reproduce please report a bug
E685: Internal error: text property above joined line not found
E340: Internal error; if you can reproduce please report a bug
E967: Text property info corrupted
E340: Internal error; if you can reproduce please report a bug
E967: Text property info corrupted
E340: Internal error; if you can reproduce please report a bug
E967: Text property info corrupted
E340: Internal error; if you can reproduce please report a bug
E967: Text property info corrupted
```

**Comment by chrisbra** [comment]
can you reproduce with `vim --clean` What filetype was this?

**Comment by twistedkento** [comment]
Yeah produced it with both `vim --clean` and opening it with `vim --clean <filename>`. 
And `set filetype?` gives me `filetype=go`

**Comment by twistedkento** [comment]
Yeah only running `:set formatoptions+=a` would probably be very bad but since I'm running `:set formatoptions=acq` described here https://github.com/vim/vim/blob/5036e698520b2c39e4df5738f026a68ba2e76fef/runtime/doc/change.txt#L1859
and if I remember correctly using `:set formatoptions=ac` should make it at least adhere to `:set comments?` setting?

**Comment by twistedkento** [comment]
Don't know where/if I read that the `auto-format` will adhere to `comments` when `formatoptions=ac` is enabled.
But here is at how it says in the help page:

https://github.com/vim/vim/blob/5036e698520b2c39e4df5738f026a68ba2e76fef/runtime/doc/change.txt#L1698-L1699

And that the part about `auto-format` only touching comments when combined with the `c` flag in formatoptions:
https://github.com/vim/vim/blob/5036e698520b2c39e4df5738f026a68ba2e76fef/runtime/doc/change.txt#L1719-L1722
